pymatgen/entries/compatibility.py

```diff
@@ -640,34 +759,224 @@ def __init__(self, compat_type="Advanced", correct_peroxide=True,
+                              "Cl2": 0.344373,
+                              "Br": 0.235039,
+                              "Hg": 0.234421,
+                              "H2O": 0.215891,
+                              }
+        self.name = "MP Aqueous free energy adjustment"
+        super().__init__()
+
+    def get_adjustments(self, entry: ComputedEntry):
         """
-        self.compat_type = compat_type
-        self.correct_peroxide = correct_peroxide
-        self.check_potcar_hash = check_potcar_hash
-        fp = os.path.join(MODULE_DIR, "MPCompatibility.yaml")
-        super().__init__(
-            [PotcarCorrection(MPRelaxSet, check_hash=check_potcar_hash),
-             GasCorrection(fp),
-             AnionCorrection(fp, correct_peroxide=correct_peroxide),
-             UCorrection(fp, MPRelaxSet, compat_type), AqueousCorrection(fp)])
+        Returns the corrections applied to a particular entry.
+
+        Args:
+            entry: A ComputedEntry object.
+
+        Returns:
+            [EnergyAdjustment]: Energy adjustments to be applied to entry.
+
+        Raises:
+            CompatibilityError if the required O2 and H2O energies have not been provided to
+            MaterialsProjectAqueousCompatibility during init or in the list of entries passed to process_entries.
+        """
+        adjustments = []
+        if not all([self.o2_energy, self.h2o_energy, self.h2o_adjustments]):
+            raise CompatibilityError("You did not provide the required O2 and H2O energies. "
+                                     "{} needs these energies in order to compute "
+                                     "the appropriate energy adjustments. Either specify the energies as arguments "
+                                     "to {}.__init__ or run process_entries on a list that includes ComputedEntry for "
+                                     "the ground state of O2 and H2O.".format(type(self).__name__, type(self).__name__))
+
+        # compute the free energies of H2 and H2O (eV/atom) to guarantee that the
+        # formationfree energy of H2O is equal to -2.4583 eV/H2O from experiments
+        # (MU_H2O from pourbaix module)
+
+        # Free energy of H2, fitted using Eq. 40 of Persson et al. PRB 2012 85(23)
+        # for this calculation ONLY, we need the DFT energy of water
+        self.h2_energy = round(
+            0.5 * ((self.h2o_energy * 3 - self.cpd_entropies["H2O"]) -
+                   (self.o2_energy - self.cpd_entropies["O2"]) -
+                   self.MU_H2O
+                   ), 6
+        )
+
+        # Free energy of H2O, fitted for consistency with the O2 and H2 energies.
+        self.fit_h2o_energy = round((2 * self.h2_energy +
+                                    (self.o2_energy - self.cpd_entropies["O2"]) +
+                                    self.MU_H2O
+                                     ) / 3,
+                                    6
+                                    )
```

Comment:
This is why mypy is unhappy. `get_adjustments` performs a check to make sure that none of the required parameters are `None` before executing the rest of the code (which would required floats), but mypy picks up the fact that these parameters could initially be equal to `None` and says you can't multiple or add `None` with floats. 

I have a hard time understanding mypy's logic in cases like this: there is no way that, for example, line 859 will execute if `self.h2o_energy` is `None`, but I still get the error.

Reply:
> I have a hard time understanding mypy's logic in cases like this: there is no way that, for example, line 859 will execute if self.h2o_energy is None, but I still get the error.

It is possible though? I can set `self.h2o_energy = None` and then call `.get_adjustments` ? This isn't using the class as intended, but it is possible -- so I think mypy is correct.

Reply:
Oh, I see the `all` statement ...

Reply:
It's possible mypy doesn't understand the `all` maybe (?) ... docs seems to suggest they use `assert x is not None` or similar (`if x is None: raise ...`). If these are required, why are they optional kwargs with None defaults anyway?

Reply:
> It's possible mypy doesn't understand the `all` maybe (?) ... docs seems to suggest they use `assert x is not None` or similar (`if x is None: raise ...`). If these values are required anyway, why do they optional kwargs with None defaults?

Those kwargs are optional to preserve flexibility, so that if a user prefers to just populate automatically by including entries for O2 and H2O when they call `process_entries`, they can do that instead of having to manually set the entries up-front.

Reply:
An alternative would be to always determine the parameters in `process_entries` from the contents of the entries, but I figured some users may have scripts or workflows in which it would not always be convenient to include H2O and O2 entries every time.

Reply:
Also, because `get_adjustments` is public, there needs to be way of handling the situation where a user instantiates the class, then tries to call `get_adjustments` before having run `process_entries`. I think there is an argument for making `get_adjustments` a private method though, to discourage that type of use. In that case, always relying on `process_entries` to populate the kwargs would make more sense.